slice-test/slice-core-test/src/test/groovy/com/cognifide/slice/core/provider/SliceModelProviderTest.groovy

```diff
@@ -210,38 +219,42 @@ class SliceModelProviderTest extends BaseSetup {
 	}
 
 	def "Get model by a null path"() {
-		def nullPath = null
+		String nullPath = null
+		setup: "Creating model for null path"
+		SimpleModel model = modelProvider.get(SimpleModel.class, nullPath);
 
-		when: "Creating model based on null path"
-		modelProvider.get(SimpleModel.class, nullPath)
+		expect: "Model should not be null"
+		Assert.assertNotNull(model)
 
-		then: "Throw ProvisionException - either path or resource should be set"
-		thrown(ProvisionException) //FIXME we should not expect ProvisionException here
+		and: "Model should have a null value for property 'prop1' set"
+		Assert.assertNull(model.getProp1())
 	}
 
 	def "Get model by a resource - non-existing (null) resource"() {
 		def nonExistingPath = "/content/nonexisting"
 
-		setup: "Get non-existing resource"
+		setup: "Creating model for path: " + nonExistingPath
 		def resource = resourceResolver.getResource(nonExistingPath)
+		def model = modelProvider.get(SimpleModel.class, resource)
 
-		when: "Creating model for non-existing (null) resource"
-		modelProvider.get(SimpleModel.class, resource);
+		expect: "Model should not be null"
+		Assert.assertNotNull(model)
 
-		then: "Throw ProvisionException - either path or resource should be set"
-		thrown(ProvisionException)
+		and: "Model should have a null value for property 'prop1' set"
+		Assert.assertNull(model.getProp1())
 	}
 
 	def "Get model resource by key and resource - non-existing (null) resource"() {
 		def nonExistingPath = "/content/nonexisting"
 
-		setup: "Get non-existing resource"
+		setup: "Creating model for path: " + nonExistingPath
```

Comment:
Seeing "Creating model for path: /content/nonexisting" in test output is not that informative as "Get non-existing resource". Use when-then as it was previously.